### PR TITLE
7-employee-factory :: Employee Factory

### PR DESCRIPTION
**Pull request by matthamil**
## Status
**Ready**

## Testing

[X] I have added passing tests for all new functionality.

[X] I have confirmed that all test prior to this pull request still PASS.

## Documentation

[X] I have fully documented features in this PR's code. Please put an `X` in the box to confirm.

[ ] This PR does **NOT** require documentation.

## Description
An `EmployeeFactory` now exists. The `EmployeeFactory` can `get(int EmployeeId)` a single employee record, or `getAll()` to return all employee records from the database.

## Resolves Issue Number

https://github.com/TeamCharles/customer-service/issues/7

## Deploy Notes

```sh
git pull --prune
git checkout 7-employee-factory
dotnet test # inside test folder
```

## Steps to Test or Reproduce

1. Run the tests yo

## Impacted Files in Application

* Factories/EmployeeFactory.cs

* EmployeeFactoryTests.cs

### CODE REVIEW

File: ReadMe.md [hunk 1]
```diff
@@ -1,81 +1,2 @@
-# Bangazon Workforce Management
```
Comment (MattKraatz) [inline ReadMe.md:1]: Could you please not override this ReadMe on your PR?

File: customer-service-tests/EmployeeFactoryTests.cs [hunk 2]
```diff
@@ -14,15 +14,35 @@ public class EmployeeFactoryTests
...
+            var employee = employeeFactory.get(id);
 
-            //var expectedType = typeof(Employee);
-            //var actualType = typeof(employee);
+            Assert.NotNull(employee);
+            Assert.True(employee.GetType() == typeof(Employee));
+            Assert.True(employee.FirstName.Length > 0);
+            Assert.True(employee.LastName.Length > 0);
+            Assert.True(employee.DepartmentId.GetType() == typeof(int));
+            Assert.True(employee.Administrator.GetType() == typeof(bool));
```
Comment (MattKraatz) [inline customer-service-tests/EmployeeFactoryTests.cs:40]: You could replace all these with `Assert.IsType<>` if you wanted to

File: src/customer-service/Factories/EmployeeFactory.cs [hunk 3]
```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Data.Sqlite;
+using customer_service.Models;
+using customer_service.Data;
+
+namespace customer_service.Factories
```
Comment (MattKraatz) [inline src/customer-service/Factories/EmployeeFactory.cs]: Do we want to segment the namespace? I guess it doesn't hurt, I just want everyone to be consistent.

**Review by MattKraatz — APPROVED** [review]
LGTM

**Comment by matthamil** [comment]
Ready for review @TeamCharles/team

File: ReadMe.md (hunk 1 above)
Comment (matthamil) [reply ReadMe.md:1]: Removed from PR

File: customer-service-tests/EmployeeFactoryTests.cs (hunk 2 above)
Comment (matthamil) [reply customer-service-tests/EmployeeFactoryTests.cs:40]: `Assert.IsType` is a few lines above this `foreach` loop. These are here to ensure that no property comes back as `null` on any Employee record.

**Review by MattKraatz — APPROVED** [review]
So phresh 🍑 🍨

**Comment by daynewright** [comment]
LGTM

**Review by Anulfo — COMMENTED** [review]
LGTM

**Comment by garrett-vangilder** [comment]
LGTM

**Comment by matthamil** [comment]
Merging!

**Pull request merged**